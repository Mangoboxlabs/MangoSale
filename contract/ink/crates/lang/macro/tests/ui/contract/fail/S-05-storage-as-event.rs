use ink_lang as ink;

#[ink::contract]
mod storage_as_event {
    #[ink(storage)]
    #[ink(event)] // We cannot have #[ink(event)] if we already have #[ink(storage)]
    pub struct StorageAsEvent {}

    impl StorageAsEvent {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
