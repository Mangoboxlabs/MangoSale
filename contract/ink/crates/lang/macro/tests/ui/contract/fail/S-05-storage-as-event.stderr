error: encountered conflicting ink! attribute argument
 --> $DIR/S-05-storage-as-event.rs:6:11
  |
6 |     #[ink(event)] // We cannot have #[ink(event)] if we already have #[ink(storage)]
  |           ^^^^^
